public/js/reducers/sources.js

```diff
@@ -205,16 +205,12 @@ function getNewSelectedSource(state, id) : ?Source {
 // (right now) to type those wrapped functions.
 type OuterState = { sources: Record<SourcesState> };
 
-function getSource(state: OuterState, id: string) {
-  return state.sources.sources.get(id);
+function getSourceByURL(sources: any, url: string) {
+  return sources.find(source => source.get("url") == url);
 }
 
-function getSourceByURL(state: OuterState, url: string) {
-  return state.sources.sources.find(source => source.get("url") == url);
-}
-
-function getSourceById(state: OuterState, id: string) {
-  return state.sources.sources.find(source => source.get("id") == id);
+function getSourceById(sources: any, id: string) {
```

Comment:
FWIW, I know you're going to revamp this PR, but wanted to point this out early. These are selectors. Selectors *always* receive state as the first argument, so I would have asked you to rework this PR. I haven't gone through it enough to understand why this needs changing, but I'm sure there's a reason, but there's probably a better way to do what you are doing. If you want to talk about it, feel free to hit me up as jlongster on irc.mozilla.org, or just ask questions in this PR. Or you can make a new PR and we can discuss it there :)